### PR TITLE
feat: table dashboard

### LINKED ISSUE #985: Dashboard for spec status
We want a dashboard that shows us where the spec and implemtnations are at, to drive priotisation of work and clearly inform all stakeholders how far we have come and what to focus on next.

Mock up: https://bl.ocks.org/jbenet/f3557b26a324823bc48807feb0a55ed2

Screenshot 2020-07-17 at 15 16 18

Sheet where the current state of the spec and impls is being aggregated: https://docs.google.com/spreadsheets/d/1XjYE18tYfSn90ZXivUFSsVhCr1Ov1XYqCrWU9mPhyzs/edit#gid=2090010128

### PR DESCRIPTION
**Pull request by hugomrdias**
This PR adds a Spec status dashboard

Screenshot 2020-07-21 at 19 04 19

todo:

- [x] documentation for the frontmatter properties

closes #985

### CODE REVIEW

File: layouts/shortcodes/dashboard-table.html [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+</script>
+{{ define "dashboard-section-children" }}
+    {{- $scratch := newScratch -}}
+    {{- $level := .level -}}
+    {{- $level = mul $level 10 -}}
+    {{- $pages := where .Section.Pages "Params.bookhidden" "ne" true -}}
+    {{- $pages2 := where $pages "Params.dashboardhidden" "ne" true -}}
+    {{- range sort $pages2 "Weight" "asc" -}}
+        {{- $level = add $level 1 -}}
+        {{- $scratch.Set "dashboard-weigth" 0 -}}
```
Comment (olizilla) [inline layouts/shortcodes/dashboard-table.html:30]:
```suggestion
        {{- $scratch.Set "dashboard-weight" 0 -}}
```

File: layouts/shortcodes/dashboard-table.html [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        {{- $scratch.Set "dashboard-audit" 0 -}}
+        {{- if .IsSection -}}
+            <tr>
+                <td style="max-width:250px;white-space: nowrap;overflow: hidden;text-overflow: ellipsis;">{{- delimit (split $level "") "." }} <a title="{{.Title}}" href="{{.Permalink}}">{{.Title}}</a></td>
+                <!-- Weight -->
+                <td>{{- (default ($scratch.Get "dashboard-weigth") .Params.dashboardWeight) -}}</td>
```
Comment (olizilla) [inline layouts/shortcodes/dashboard-table.html:36]:
```suggestion
                <td>{{- (default ($scratch.Get "dashboard-weight") .Params.dashboardWeight) -}}</td>
```

File: layouts/shortcodes/dashboard-table.html [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                {{- if eq .Params.dashboardState "incorrect" -}}
+                    {{ $color = "#ff0010" }}
+                {{- else if eq .Params.dashboardState "wip" -}}
+                    {{ $color = "#ff6f00" }}
+                {{- else if eq .Params.dashboardState "incomplete" -}}
+                    {{ $color = "#ffef00" }}
+                {{- else if eq .Params.dashboardState "stable" -}}
+                    {{ $color = "#90ff00" }}
+                {{- end -}}
+                <td style="background-color:{{- $color -}};color: black">{{- (default "N/A" (humanize .Params.dashboardState)) -}}</td>
```
Comment (olizilla) [inline layouts/shortcodes/dashboard-table.html:48]: WIP is acronym, all caps 🧢 

```suggestion
                <td style="background-color:{{- $color -}};color: black">
                  {{- if eq .Params.dashboardState "wip" -}}
                    WIP
                  {{- else -}}
                    {{- (default "N/A" (humanize .Params.dashboardState)) -}}
                  {{- end -}}
                </td>
```

File: layouts/shortcodes/dashboard-table.html [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            {{- if eq .Params.dashboardState "incorrect" -}}
+                {{ $color = "#ff0010" }}
+            {{- else if eq .Params.dashboardState "wip" -}}
+                {{ $color = "#ff6f00" }}
+            {{- else if eq .Params.dashboardState "incomplete" -}}
+                {{ $color = "#ffef00" }}
+            {{- else if eq .Params.dashboardState "stable" -}}
+                {{ $color = "#90ff00" }}
+            {{- end -}}
+            <td style="background-color:{{- $color -}};color: black">{{- (default "N/A" (humanize .Params.dashboardState)) -}}</td>
```
Comment (olizilla) [inline layouts/shortcodes/dashboard-table.html:85]: How about pulling the colors into css classes for each state, so we can drop the if/else color blocks
```css
.bg-incorrect { background-color: #ff0010 }
.bg-wip { background-color: #ff6f00 }
.bg-incomplete { background-color: #ffef00 }
.bg-stable { background-color: #90ff00 }
.black { color: #111 } 
```
```suggestion
            <td class="{{- .Params.dashboardState -}} black">{{- (default "N/A" (humanize .Params.dashboardState)) -}}</td>
```

**Comment by olizilla** [comment]
I love space and breathing room very much... but I think for this table we might actually want information density to be the thing... like _"I want to see as much of the table as possible at once"_. Normally that would be a ux _no no_, but as the cells only confer the state of a enum rather than arbitrary precision numbers, it's not so bad... something like 0 vertical padding and font size 12px...

Screenshot 2020-07-22 at 09 50 13

**Comment by daviddias** [comment]
Recommendation for the headers

image

**Comment by daviddias** [comment]
Does the weight need to be here? 
image

**Comment by daviddias** [comment]
Btw, with this, it should be easy to get the 3 metric as well at the top, correct (Progress, Risk & Consequence)? (perhaps at least just Progress?)

**Comment by daviddias** [comment]
> Do you think we still need to use Theory Audit and Compliance Tests ?

Yes, using consisting terminology in a project where every key word is overloaded is essential for clarity and avoiding misinformation

**Pull request merged**